src/Lean/Meta/Tactic/SplitIf.lean

```diff
@@ -56,23 +56,25 @@ def discharge? (useDecide := false) : Simp.Discharge := fun prop => do
          else
            return none
 
-/-- Return the condition of an `if` expression to case split. -/
-partial def findIfToSplit? (e : Expr) : Option Expr :=
+/-- Return the condition (and decidability instance when available) of an `if` expression to case split. -/
+partial def findIfToSplit? (e : Expr) : Option (Expr × Option Expr) :=
   if let some iteApp := e.find? fun e => (e.isIte || e.isDIte) && !(e.getArg! 1 5).hasLooseBVars then
     let cond := iteApp.getArg! 1 5
+    -- If the decidability instance has loose bound variables then we fall back on using classical axioms
+    let inst := if (e.getArg! 2 5).hasLooseBVars then none else some <| iteApp.getArg! 2 5
```

Comment:
`e` should be `iteApp` here, no?

If so, please expand the test coverage to the point that this would have been caught. :-)

In either case, there should be a test that Mario's example succeeds still.

Reply:
I added Mario's example in the tests. 

I think that test would also catch the typo since `e.getArg! 2 5` would be `default` and wouldn't have loose bound variables but `iteApp.getArg! 2 5` would have some. I have a hard time thinking of a test that would more specifically catch that typo.